### PR TITLE
fix(analytics): setUserId accepts null values

### PR DESCRIPTION
**Pull request by simondaigre**
Since I've implemented setUserId on a web project, I have less unique visitors than expected. I have some doubts of the implementation of setUserId, which currently set an empty string if we pass a null value.

According to the Firebase JS docs, null is accepted : https://firebase.google.com/docs/reference/js/analytics.md?hl=fr#setuserid_86d82f6
And iOS / Android docs says that userId must be a non empty string : https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics#public-void-setuserid-string-id https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics#public-void-setuserid-string-id

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] The changes have been tested successfully.
- [ ] A changeset has been created (`npm run changeset`).
- [ ] I have read and followed the [pull request guidelines](https://capawesome.io/contributing/pull-requests/).

### CODE REVIEW

**Review by robingenz — DISMISSED** [review]
Good catch! I'm not sure why we use an empty string as fallback.

**Comment by robingenz** [comment]
@simondaigre Please create a changeset by running the following command:

```
npm run changeset
```

**Comment by simondaigre** [comment]
@robingenz done!

File: package-lock.json
Comment (robingenz) [inline package-lock.json:1]: Please revert the changes in this file. You can use this command: `git checkout origin/main package-lock.json`

**Review by robingenz — CHANGES_REQUESTED** [review]

**Comment by robingenz** [comment]
@simondaigre Thank you for your contribution!
